astropy/visualization/time.py

```diff
@@ -0,0 +1,261 @@
+# -*- coding: utf-8 -*-
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+import numpy as np
+
+from datetime import datetime
+
+from astropy.time import Time
+from astropy import units as u
+
+__all__ = ['time_support']
+
+__doctest_requires__ = {'time_support': ['matplotlib']}
+
+UNSUPPORTED_FORMATS = ('datetime', 'datetime64')
+YMDHMS_FORMATS = ('fits', 'iso', 'isot', 'yday')
+STR_FORMATS = YMDHMS_FORMATS + ('byear_str', 'jyear_str')
+
+
+def time_support(scale=None, format=None, simplify=True):
+    """
+    Enable support for plotting `astropy.time.Time` instances in
+    matplotlib.
+
+    May be (optionally) used with a ``with`` statement.
+
+      >>> import matplotlib.pyplot as plt
+      >>> from astropy import units as u
+      >>> from astropy import visualization
+      >>> with visualization.time_support():  # doctest: +IGNORE_OUTPUT
+      ...     plt.figure()
+      ...     plt.plot(Time(['2016-03-22T12:30:31', '2016-03-22T12:30:38', '2016-03-22T12:34:40']))
+      ...     plt.draw()
+
+    Parameters
+    ----------
+    format : str, optional
```

Comment:
`scale` should come before `format` to match the function signature.

Reply:
Fixed, and I also made the function signature have keyword-only arguments.